xarray/backends/zarr.py

```diff
@@ -440,6 +458,12 @@ def open_store_variable(self, name, zarr_array):
             zarr_array, DIMENSION_KEY, try_nczarr
         )
         attributes = dict(attributes)
+
+        # TODO: how to properly handle 'filters' for v3 stores
```

Comment:
What needs working to "properly handle" filters?

Reply:
@grlee77 - do you recall what you meant here?

Reply:
'filters' is not part of the core v3 spec, so exactly how it should work hasn't been specified yet. There is a placeholder [here](https://zarr-specs.readthedocs.io/en/latest/extensions/array-extensions/filters/v1.0.html) for defining an extension to the spec for it.

For zarr-python, it should still work as for v2, it is just that since there is no longer a `filters` field in the v3 metadata I [chose to store it under an "attributes" section instead](https://github.com/zarr-developers/zarr-python/blob/b02040471f9b7e3e1aefd4f6b90b52c76efe32b8/zarr/core.py#L284-L292). I just did that so that existing use cases in the Zarr test suite would work for v3, but it is possible the protocol extension linked above, if developed further, may change where the metadata should be stored.

Reply:
I think maybe at the time I wrote this comment, I had not put that attributes workaround for the filters in place, but I am not sure.

Reply:
I just had a chat about this with Zarr developers including @jstriebel. Recent changes to the Zarr V3 spec make this workaround unnecessary. The [current](https://zarr-specs.readthedocs.io/en/latest/core/v3.0.html) version says:

> comparison with spec v2: ... the separate `filters` and `compressor` fields have been combined into the single `codecs` field

So the correct thing to for V3 stores is to just use the `codecs` field for encoding. The workaround in zarr of pushing filters to array attributes should be removed. I'll open a PR for that.

Reply:
Thanks adding some context here. I added a comment to remove the workaround once https://github.com/zarr-developers/zarr-python/issues/1269 is resolved. Until then, I suggest we leave the work around in place.